custom/enikshay/integrations/utils.py

```diff
@@ -61,11 +71,18 @@ def is_valid_voucher_submission(voucher_case):
     return is_valid_person_submission(person_case)
 
 
-def is_valid_test_submission(test_case):
+def is_valid_test_case_submission(test_case):
+    """
+    this checks if the test case is submitted by a valid/real user
+    """
+    person_case = get_person_case_from_test(test_case.domain, test_case.case_id)
+    if is_invalid_person_submission(person_case):
+        return False
+
     try:
         lab_referral_case = get_lab_referral_from_test(test_case.domain, test_case.get_id)
     except ENikshayCaseNotFound:
-        return False
+        return True
```

Comment:
Why is this reversed?

Reply:
Reverted the logic of method and its name. It was pretty confusing. 
If you see the [commit](https://github.com/dimagi/commcare-hq/pull/18227/commits/245e92790329ebedfc32771761901467060e51a7)

Reply:
I think this should still be `False`? I think the logic before might have been wrong.

Reply:
In case the lab referral case was not found the logic earlier would return False, which means that its not a test case i.e its a real case. That probably was done to not stop processing in case the method failed to determine if its a test case or not.

So now since the expectation on return value is reversed, `False` -> `True`